Framework/DataObjects/src/EventWorkspace.cpp

```diff
@@ -590,8 +590,12 @@ void EventWorkspace::resetAllXToSingleBin() {
 
   // Sanitize TOF min/max to ensure it always passes HistogramX validation.
   // They would be invalid when number of events are 0 or 1, for example.
-  tofmin = tofmin < tofmax ? tofmin : 0;
-  tofmax = std::max(tofmax, tofmin + std::numeric_limits<double>::min());
+  if (tofmin > tofmax) {
+    tofmin = 0;
+    tofmax = std::numeric_limits<double>::min();
+  } else if (tofmin == tofmax) {
+    tofmax += std::numeric_limits<double>::min();
+  }
```

Comment:
This isn't quite the same because this change would alter tofmax in the case `tofmax > tofmin`, when it should stay unaltered.

Also, somewhat odd:
```c++ 
std::numeric_limits<double>::min() == 2.22507e-308
```
... but:
```c++
std::nexttoward(0.0, std::numeric_limits<double>::max()) == 4.94066e-324
```

Reply:
`std::nexttoward` includes subnornal values, so

```cpp
std::nexttoward(0.0, std::numeric_limits<double>::max()) == std::numeric_limits<double>::denorm_min()
```